### ISSUE TITLE
run desktop user pref commands as vagrant

### ISSUE THREAD

**Issue by stefco** [stefco/geco_vm#53]

**Comment by stefco** [comment]
Implemented in 8b635e7

**Comment by stefco** [comment]
not actually working...

**Comment by stefco** [comment]
http://askubuntu.com/questions/135573/gconf-error-no-d-bus-daemon-running-how-to-reinstall-or-fix

**Comment by stefco** [comment]
```
    virtualbox-iso: + sleep 30
    virtualbox-iso: + gsettings set org.gnome.desktop.screensaver lock-enabled false
    virtualbox-iso:
    virtualbox-iso: ** (process:1951): WARNING **: Command line `dbus-launch --autolaunch=3e3a0b92ee0a77b8434ebc600000008f --binary-syntax --close-stderr' exited with non-zero exit status 1: Autolaunch error: X11 initialization failed.\n
    virtualbox-iso:
    virtualbox-iso: ** (process:1951): WARNING **: Command line `dbus-launch --autolaunch=3e3a0b92ee0a77b8434ebc600000008f --binary-syntax --close-stderr' exited with non-zero exit status 1: Autolaunch error: X11 initialization failed.\n
    virtualbox-iso: + gsettings set com.canonical.Unity.Launcher favorites '['\''nautilus-home.desktop'\'', '\''/home/vagrant/.medm_launcher/medm-lho.desktop'\'', '\''/home/vagrant/.medm_launcher/medm-llo.desktop'\'', '\''gnome-terminal.desktop'\'', '\''firefox.desktop'\'', '\''libreoffice-writer.desktop'\'', '\''libreoffice-calc.desktop'\'', '\''libreoffice-impress.desktop'\'']'
    virtualbox-iso:
    virtualbox-iso: ** (process:1955): WARNING **: Command line `dbus-launch --autolaunch=3e3a0b92ee0a77b8434ebc600000008f --binary-syntax --close-stderr' exited with non-zero exit status 1: Autolaunch error: X11 initialization failed.\n
    virtualbox-iso:
    virtualbox-iso: ** (process:1955): WARNING **: Command line `dbus-launch --autolaunch=3e3a0b92ee0a77b8434ebc600000008f --binary-syntax --close-stderr' exited with non-zero exit status 1: Autolaunch error: X11 initialization failed.\n
==> virtualbox-iso: Gracefully halting virtual machine...
==> virtualbox-iso: Preparing to export machine...
    virtualbox-iso: Deleting forwarded port mapping for the communicator (SSH, WinRM, etc) (host port 2788)
==> virtualbox-iso: Exporting virtual machine...
    virtualbox-iso: Executing: export geco-vm-gui --output packer-geco-vm-gui-virtualbox/geco-vm-gui.ovf
==> virtualbox-iso: Unregistering and deleting virtual machine...
==> virtualbox-iso: Running post-processor: vagrant
==> virtualbox-iso (vagrant): Creating Vagrant box for 'virtualbox' provider
    virtualbox-iso (vagrant): Copying from artifact: packer-geco-vm-gui-virtualbox/geco-vm-gui-disk1.vmdk
    virtualbox-iso (vagrant): Copying from artifact: packer-geco-vm-gui-virtualbox/geco-vm-gui.ovf
    virtualbox-iso (vagrant): Renaming the OVF to box.ovf...
    virtualbox-iso (vagrant): Compressing: Vagrantfile
    virtualbox-iso (vagrant): Compressing: box.ovf
    virtualbox-iso (vagrant): Compressing: geco-vm-gui-disk1.vmdk
    virtualbox-iso (vagrant): Compressing: metadata.json
==> virtualbox-iso: Running post-processor: atlas
    virtualbox-iso (atlas): Uploading artifact (2007968845 bytes)
Build 'virtualbox-iso' finished.
```

**Comment by stefco** [comment]
one hacky way to do it:

http://askubuntu.com/questions/462143/where-are-the-startup-scripts-for-unity-desktop

these are autostart .desktop programs; can just make one to launch whenever the system turns on. could even, as part of this script, delete the file with something like

``` bash
sleep 15
rm /path/to/startup/script
```

which is admittedly a terrible hacky way to get this setup the first time while still allowing users to mess around themselves.

**Comment by stefco** [comment]
maybe the `desktop-user-prefs.sh` script should be renamed to `geco_reset_desktop_prefs.sh` and placed in `~/bin`

**Comment by stefco** [comment]
- [x] probably should `chmod +x` both of those guys

**Comment by stefco** [comment]
or does the .desktop file go in /usr/share/xsessions?

http://askubuntu.com/questions/30931/how-do-i-make-a-program-auto-start-every-time-i-log-in

**Comment by stefco** [comment]
none of this works

**Comment by stefco** [comment]
AH. looks like i need X-window forwarding. get the exact same error when using `vagrant ssh` without X-window forwarding:

```
$ gsettings set com.canonical.Unity.Launcher favorites "['nautilus-home.desktop', '/home/vagrant/.medm_launcher/medm-lho.desktop', '/home/vagrant/.medm_launcher/medm-llo.desktop', 'gnome-terminal.desktop', 'firefox.desktop', 'libreoffice-writer.desktop', 'libreoffice-calc.desktop', 'libreoffice-impress.desktop']"

** (process:3124): WARNING **: Command line `dbus-launch --autolaunch=d093c6b1492b8ede6085a1d70000008f --binary-syntax --close-stderr' exited with non-zero exit status 1: Autolaunch error: X11 initialization failed.\n

** (process:3124): WARNING **: Command line `dbus-launch --autolaunch=d093c6b1492b8ede6085a1d70000008f --binary-syntax --close-stderr' exited with non-zero exit status 1: Autolaunch error: X11 initialization failed.\n
```

**Comment by stefco** [comment]
looks v useful:

http://askubuntu.com/questions/276509/change-gsettings-without-running-x-and-unity

**Comment by stefco** [comment]
indeed, adding `dbus-launch --exit-with-session` as suggested in the previous link does make it work on reboot.

**Comment by stefco** [comment]
fixed for good (in a non-hacky way) in f2a6091
